ScrollOfResurrection.lua

```diff
@@ -0,0 +1,741 @@
+            chatHandler:SendSysMessage("Admin/GM Syntax: .forcebindsor $resurrected_account $recruiter same as .bindsor but ignores past binds. Previously unbound, succesful or timed out doesn't matter.")
+        end
+        SOR_cleanup()
+        return false
+
+
+    elseif commandArray[1] == "sor" then
+
+        if player ~= nil then
+            if SOR_checkAbuse(player:GetAccountId()) == true then
+                local resurrectedId
+                player:KickPlayer()
+                for index, value in pairs(SOR_recruiterAccount) do
+                    if value == player:GetAccountId() then
+                        resurrectedId = index
+                    end
+                end
+                if SOR_kickCounter[resurrectedId] == nil then
+                    SOR_kickCounter[resurrectedId] = 1
+                else
+                    SOR_kickCounter[resurrectedId] = SOR_kickCounter[resurrectedId] + 1
+                    CharDBExecute('UPDATE `'..Config.customDbName..'`.`scroll_of_resurrection_links` SET `kick_counter` = '..SOR_kickCounter[resurrectedId]..' WHERE `account_id` = '..resurrectedId..';')
+                    if Config.printErrorsToConsole == 1 then PrintError("SOR: account id "..player:GetAccountId().." was kicked because of too many .sor commands.") end
+                end
+            end
+        end
+
+        -- list all accounts resurrected by this account
+        if commandArray[2] == "list" then
+
+            if player == nil then
+                chatHandler:SendSysMessage("'.sor list' is not meant to be used from the console.")
+                return false
+            end
+
+            -- print all resurrected accounts bound to this account by charname
+            local idList
+            for index, value in pairs(SOR_recruiterAccount) do
+                if value == player:GetAccountId() then
+                    if SOR_timeStamp[index] > 1 then
+                        if idList == nil then
+                            idList = index
+                        else
+                            idList = idList.." "..index
+                        end
+                    end
+                end
+            end
+            if idList == nil then
+                idList = "none"
+            end
+
+
+            chatHandler:SendSysMessage("Your current resurrected accounts are: "..idList)
+            SOR_cleanup()
+            return false
+
+        elseif commandArray[2] == "summon" then
+            if player == nil then
+                chatHandler:SendSysMessage("'.sor summon' is not meant to be used from the console.")
+                return false
+            end
+
+            if commandArray[3] == nil then
+                chatHandler:SendSysMessage("'.sor summon' requires the name of the target. Use '.sor summon $Name'")
+                return false
+            else
+                commandArray[3] = commandArray[3]:gsub("^%l", string.upper)
+            end
```

Comment:
the else is not needed in this case, because after the `return` the interpreter will not go in the next instructions
```suggestion
            if commandArray[3] == nil then
                chatHandler:SendSysMessage("'.sor summon' requires the name of the target. Use '.sor summon $Name'")
                return false
            end
            
            commandArray[3] = commandArray[3]:gsub("^%l", string.upper)
```

Reply:
That's a misunderstanding. The `return false` isn't executed, if `commandArray[3]` doesn't equal nil. It needs to be formated then.
This is just a nil check, making sure the variable isn't empty, preventing further execution, and throwing an error.